lib/rake/extensiontask.rb

```diff
@@ -278,6 +278,16 @@ def define_native_tasks(for_platform = nil, ruby_ver = RUBY_VERSION, callback =
             "< #{ruby_api_version(sorted_ruby_versions.last).succ}.dev"
           ]
 
+          # set rubygems version constraints
+          if spec.platform.os == "linux" && !spec.platform.version.nil?
+            if spec.required_rubygems_version == Gem::Requirement.default
+              spec.required_rubygems_version = [">= 3.3.22"]
+            else
+              spec.required_rubygems_version = Gem::Requirement.new(gem_spec.required_rubygems_version)
+              spec.required_rubygems_version.concat([">= 3.3.22"])
```

Comment:
Can we simplify this?

```suggestion
              spec.required_rubygems_version = spec.required_rubygems_version.requirements + [">= 3.3.22"]
```

Reply:
`Gem::Requirements` does not support the `#+` method. See https://docs.ruby-lang.org/en//3.2/Gem/Requirement.html for supported methods. This is the simplest form I can think of.

Reply:
I think that `spec.required_rubygems_version.requirements` is an `Array` not a `Gem::Requirements`.

Reply:
I've changed this to

```ruby
spec.required_rubygems_version = Gem::Requirement.new(gem_spec.required_rubygems_version, ">= 3.3.22")
```

@kou What you suggest doesn't work. `requirements` is an array, but `Gem::Specification#required_rubygems_version=` does not accept it as an argument:

```ruby
spec = Gem::Specification.new do |s|
  s.name = 'my_gem'
  s.platform = Gem::Platform::RUBY
  s.extensions = ['ext/somegem/extconf.rb']
  s.required_rubygems_version = "!= 1.0.0"
end

spec.required_rubygems_version.requirements + [">= 3.3.22"]
# => [["!=", Gem::Version.new("1.0.0")], ">= 3.3.22"]

spec.required_rubygems_version = spec.required_rubygems_version.requirements + [">= 3.3.22"]
# /home/flavorjones/.rbenv/versions/3.2.3/lib/ruby/3.2.0/rubygems/requirement.rb:107:in `parse': Illformed requirement ["!="] (Gem::Requirement::BadRequirementError)
# 
#       raise BadRequirementError, "Illformed requirement [#{obj.inspect}]"
#             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
# 	from /home/flavorjones/.rbenv/versions/3.2.3/lib/ruby/3.2.0/rubygems/requirement.rb:139:in `block in initialize'
# 	from /home/flavorjones/.rbenv/versions/3.2.3/lib/ruby/3.2.0/rubygems/requirement.rb:139:in `map!'
# 	from /home/flavorjones/.rbenv/versions/3.2.3/lib/ruby/3.2.0/rubygems/requirement.rb:139:in `initialize'
# 	from /home/flavorjones/.rbenv/versions/3.2.3/lib/ruby/3.2.0/rubygems/requirement.rb:64:in `new'
# 	from /home/flavorjones/.rbenv/versions/3.2.3/lib/ruby/3.2.0/rubygems/requirement.rb:64:in `create'
# 	from /home/flavorjones/.rbenv/versions/3.2.3/lib/ruby/3.2.0/rubygems/specification.rb:702:in `required_rubygems_version='
# 	from /home/flavorjones/bar.rb:15:in `<main>'
```

Reply:
I think that `Gem::Requirement` could be improved in two ways that would make this feature simpler to implement:

1. `#dup` should make a deep copy of `@requirements`. It does not, and so I am calling `Gem::Requirement.new(gem_spec.required_rubygems_version)` to duplicate it.
2. `#concat` could return `self`. It does not, and so I am doing this across two statements.

If those things were fixed, this could become

```ruby
spec.required_rubygems_version = spec.required_rubygems_version.dup.concat(">= 3.3.22")
```

Reply:
Oh. I see.